src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -34,7 +34,7 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
         private readonly IPdbSourceDocumentLogger? _logger;
 
         private readonly Dictionary<string, ProjectId> _assemblyToProjectMap = new();
-        private readonly Dictionary<string, (DocumentId documentId, Encoding encoding)> _fileToDocumentMap = new();
+        private readonly Dictionary<string, SourceDocumentInfo> _fileToDocumentMap = new();
```

Comment:
```suggestion
        private readonly Dictionary<string, SourceDocumentInfo> _fileToDocumentInfoMap = new();
```

Since you have multiple (similarly) named types?